flexget/components/ftp/sftp_client.py

```diff
@@ -0,0 +1,423 @@
+                       entry_accumulator)
+
+    def get_unknown_handler(self) -> Callable[[str], None]:
+        """
+        Builds an unknown node handler suitable for use with pysftp.Connection.walktree
+        """
+        return partial(Handlers.handle_unknown, self._logger)
+
+    def get_null_handler(self) -> Callable[[str], None]:
+        """
+        Builds a noop node handler suitable for use with pysftp.Connection.walktree
+        """
+        return partial(Handlers.null_node_handler, self._logger)
+
+
+class Handlers:
+    @classmethod
+    def handle_file(cls, sftp: 'pysftp.Connection', logger: logging.Logger, prefix,get_size,
+                    entry_accumulator, path):
+        """
+        File node handler. Adds a file entry to entry_accumulator.
+
+        :param sftp: A pysftp.Connection object
+        :param logger: a logger object
+        :param prefix: SFTP URL prefix
+        :param get_size: boolean indicating whether to compute the size of each file
+        :param entry_accumulator: a list in which to store entries
+        :param path: path to handle
+        """
+        size_handler = partial(cls._file_size, sftp)
+        entry = cls._get_entry(sftp, logger, prefix, size_handler, get_size, path)
+        entry_accumulator.append(entry)
+
+    @classmethod
+    def handle_directory(cls, sftp: 'pysftp.Connection', logger: logging.Logger, prefix: str, get_size: bool,
+                         files_only: bool, entry_accumulator: List, path: str):
+        """
+        Directory node handler. Adds a directory entry to entry_accumulator.
+
+        :param sftp: A pysftp.Connection object
+        :param logger: a logger object
+        :param prefix: SFTP URL prefix
+        :param get_size: boolean indicating whether to compute the size of each directory
+        :param files_only: Boolean indicating whether to skip directories
+        :param entry_accumulator: a list in which to store entries
+        :param path: path to handle
+        """
+        if files_only:
+            return
+
+        dir_size: Callable[[str], int] = partial(cls._dir_size, sftp)
+        entry: Entry = cls._get_entry(sftp, logger, prefix, dir_size, get_size, path)
+        entry_accumulator.append(entry)
+
+    @staticmethod
+    def handle_unknown(logger: logging.Logger, path: str):
+        """
+        Handler for unknown nodes; logs a warning.
+
+        :param logger: a logger object
+        :param path: path to handle
+        """
+        logger.warning('Skipping unknown file: {}', path)
+
+    @staticmethod
+    def null_node_handler(logger: logging.Logger, path: str):
+        """
+        Generic noop node handler
+
+        :param logger: a logger object
+        :param path: path to handle
+        :return:
+        """
+        logger.debug('null handler called  for {}', path)
+
+    @staticmethod
+    def _get_entry(sftp: 'pysftp.Connection', logger: logging.Logger, prefix: str,
```

Comment:
why make this a staticmethod and pass `sftp` instead of just using `self.sftp` here?